blocks/api/parser.js

```diff
@@ -88,6 +53,73 @@ export function asType( value, type ) {
 }
 
 /**
+ * Returns an hpq matcher given a source object
+ *
+ * @param  {Object}   sourceConfig Attribute Source object
+ * @return {Function}              hpq Matcher
+ */
+export function matcherFromSource( sourceConfig ) {
+	switch ( sourceConfig.source ) {
+		case 'attribute':
+			return attr( sourceConfig.selector, sourceConfig.attribute );
+		case 'property':
+			return prop( sourceConfig.selector, sourceConfig.property );
+		case 'html':
+			return html( sourceConfig.selector );
+		case 'text':
+			return text( sourceConfig.selector );
+		case 'children':
+			return children( sourceConfig.selector );
+		case 'node':
+			return node( sourceConfig.selector );
+		case 'query':
+			const subMatchers = keys( sourceConfig.query ).reduce( ( memo, key ) => {
+				memo[ key ] = matcherFromSource( sourceConfig.query[ key ] );
+				return memo;
+			}, {} );
+
+			return query( sourceConfig.selector, subMatchers );
+		default:
+			// eslint-disable-next-line no-console
+			console.error( `Unkown source type "${ sourceConfig.source }"` );
+	}
+}
+
+/**
+ * Given an attribute key, an attribute's schema, a block's raw content and the commentAttributes
+ * returns the attribute value depending on its source definition of the given attribute key
+ *
+ * @param  {string} attributeKey        Attribute key
+ * @param  {Object} attributeSchema     Attribute's schema
+ * @param  {string} rawContent          Block's raw content
+ * @param  {Object} commentAttributes   Block's comment attributes
+ *
+ * @return {*}                          Attribute value
+ */
+export function getBlockAttribute( attributeKey, attributeSchema, rawContent, commentAttributes ) {
+	let value;
+	switch ( attributeSchema.source ) {
+		case 'comment':
+			value = commentAttributes ? commentAttributes[ attributeKey ] : undefined;
+			break;
+		case 'attribute':
+		case 'property':
+		case 'html':
+		case 'text':
+		case 'children':
+		case 'node':
+		case 'query':
+		{
```

Comment:
Do you need the block scoping here?

Reply:
Yes, since we have the `const` but we could avoid it.